src/uct/ib/base/ib_device.h

```diff
@@ -151,8 +152,80 @@ typedef struct uct_ib_device_spec {
 } uct_ib_device_spec_t;
 
 
+/* Calculate real GIDs len. Can be either 16 (RoCEv1 or RoCEv2/IPv6)
+ * or 4 (RoCEv2/IPv4). This len is used for packets filtering by DGIDs.
+ *
+ * According to Annex17_RoCEv2 (A17.4.5.2):
+ * "The first 40 bytes of user posted UD Receive Buffers are reserved for the L3
+ * header of the incoming packet (as per the InfiniBand Spec Section 11.4.1.2).
+ * In RoCEv2, this area is filled up with the IP header. IPv6 header uses the
+ * entire 40 bytes. IPv4 headers use the 20 bytes in the second half of the
+ * reserved 40 bytes area (i.e. offset 20 from the beginning of the receive
+ * buffer). In this case, the content of the first 20 bytes is undefined." */
+static UCS_F_ALWAYS_INLINE
+unsigned uct_ib_gid_calc_len(const union ibv_gid *gid)
```

Comment:
will it be used for any transport except the UD?

Reply:
similar filter has to be implemented in DC

Reply:
is it really needed for any gid, or just grh which arrives in the packet?

Reply:
now it's used in `uct_ib_gid_str` for any GID